data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/GetNewBlindedBlock.java

```diff
@@ -26,29 +27,88 @@
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_VALIDATOR;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_VALIDATOR_REQUIRED;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
 import io.javalin.http.Context;
-import io.javalin.http.Handler;
 import io.javalin.plugin.openapi.annotations.HttpMethod;
 import io.javalin.plugin.openapi.annotations.OpenApi;
 import io.javalin.plugin.openapi.annotations.OpenApiContent;
 import io.javalin.plugin.openapi.annotations.OpenApiParam;
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.Optional;
+import java.util.function.Function;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
 import tech.pegasys.teku.api.DataProvider;
 import tech.pegasys.teku.api.ValidatorDataProvider;
 import tech.pegasys.teku.api.response.v1.validator.GetNewBlindedBlockResponse;
-import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
-import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.beaconrestapi.SchemaDefinitionCache;
+import tech.pegasys.teku.bls.BLSSignature;
+import tech.pegasys.teku.infrastructure.http.RestApiConstants;
+import tech.pegasys.teku.infrastructure.json.types.CoreTypes;
+import tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.json.types.SerializableOneOfTypeDefinitionBuilder;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.json.types.StringValueTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.ParameterMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.datastructures.blocks.BeaconBlock;
 
-public class GetNewBlindedBlock extends GetNewBlock implements Handler {
+@SuppressWarnings("unused")
+public class GetNewBlindedBlock extends MigratingEndpointAdapter {
   private static final String OAPI_ROUTE = "/eth/v1/validator/blinded_blocks/:slot";
   public static final String ROUTE = routeWithBracedParameters(OAPI_ROUTE);
+  private final ValidatorDataProvider provider;
 
-  public GetNewBlindedBlock(final DataProvider dataProvider, final JsonProvider jsonProvider) {
-    super(dataProvider, jsonProvider);
+  private static final ParameterMetadata<UInt64> PARAM_SLOT =
+      new ParameterMetadata<>(
+          SLOT,
+          CoreTypes.UINT64_TYPE.withDescription(
+              "The slot for which the block should be proposed."));
+
+  private static final StringValueTypeDefinition<BLSSignature> SIGNATURE_TYPE =
```

Comment:
nit: Is it worth splitting this out into an `EthereumTypes` class now even if it's only in the beacon rest API package for now just to avoid having it pop up in a lot of different APIs as we migrate?

Reply:
we could, just currently i didnt really have a reason to, even though it does seem obvious it'll happen...